apps/web/src/client/components/files-list/files-list.module.js

```diff
@@ -43,7 +56,42 @@ const initFilesListModule = () => {
 		selectedFilesNumber.textContent = `${(checkedFiles || []).length}`;
 	};
 
+	/**
+	 * Download many files at once
+	 *
+	 * @param {*} clickEvent
+	 */
+	const bulkDownload = (clickEvent) => {
+		clickEvent.preventDefault();
+
+		/** @type {NodeListOf<HTMLInputElement>} */
+		const checkedFiles = document.querySelectorAll('input[name="selectedFilesIds[]"]:checked');
+
+		if (checkedFiles.length === 0) {
+			topHook.innerHTML = buildErrorBanner('Select documents to download');
+			topHook.scrollIntoView();
+
+			return;
+		}
+
+		for (const selectedCheckbox of checkedFiles) {
+			const { value: selectedGuid } = selectedCheckbox;
+
+			/** @type {HTMLElement | null} */
+			const downloadLink = document.querySelector(`a[data-action='download-${selectedGuid}']`);
+
+			if (downloadLink) {
+				// system interrupts download requests if they all happen at the very same moment
```

Comment:
I tested this with a big file and still works, which means that it's ok to send a download request when other files are still being downloaded. Important thing is not to send all the requests at once.